backend/tests/test_b055_llm_worker_stubs.py

```
@@ -164,6 +169,90 @@ async def test_llm_explanation_stub_writes_api_call(test_tenant):
+            )
+        ).scalars().one()
+        assert monthly.total_calls == 1, "Idempotency failed: monthly costs double-counted"
+
+
+@pytest.mark.asyncio
+async def test_llm_explanation_retry_preserves_request_id_when_omitted(test_tenant, monkeypatch):
+    _assert_postgres_engine()
+    payload = {"stub": True}
+    captured = {}
+
+    def _fake_retry(*, kwargs=None, **_):
```

Ruff (0.14.11)

Missing return type annotation for private function `_fake_retry`

Add return type annotation: `NoReturn`

(ANN202)